src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -28,7 +28,8 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
-                    shallowClone(repo.shallow?:false)
+                    //shallow clone is the default
+                    shallowClone(repo.shallow!=false?true:false)
```

Comment:
null *should* evaluate to true here if shallow clone is to be the default. (I try to avoid ternaries like this because the intent can get confusing.)